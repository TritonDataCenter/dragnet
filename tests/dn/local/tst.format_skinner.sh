#!/bin/bash

#
# tst.format_skinner.sh: tests "scan", "index", and "query" operations on
# skinner-style points as input data.
#

set -o errexit
. $(dirname $0)/../common.sh

function trace
{
	echo "#" "$@"
	"$@"
}

tmpfile="/var/tmp/$(basename $0).$$"
tmpfile2="$tmpfile.2"
echo "using tmpfiles \"$tmpfile\" and \"$tmpfile2\"" >&2

dn_clear_config
dn datasource-add stdin --path=/dev/stdin
dn datasource-add stdin-skinner --path=/dev/stdin --data-format=json-skinner

# Test points with no fields
dn scan --points stdin < $DN_DATADIR/2014/05-01/one.log > $tmpfile

cat $tmpfile | trace dn scan stdin-skinner
cat $tmpfile $tmpfile | trace dn scan stdin-skinner
cat $tmpfile $tmpfile $tmpfile | trace dn scan stdin-skinner

# Test points with a couple of fields
dn scan --points -b req.method,res.statusCode stdin \
    < $DN_DATADIR/2014/05-01/one.log > $tmpfile
dn scan -b req.method stdin < $DN_DATADIR/2014/05-01/one.log
cat $tmpfile $tmpfile $tmpfile | trace dn scan stdin-skinner
cat $tmpfile $tmpfile $tmpfile | trace dn scan stdin-skinner -b req.method 

# Test indexes
echo "building index"
cat $tmpfile $tmpfile $tmpfile > $tmpfile2
mv $tmpfile2 $tmpfile
dn datasource-add test_input --path=$tmpfile --data-format=json-skinner \
    --index-path=$tmpfile2 
dn metric-add test_input total
dn metric-add test_input -b req.method by_method
dn build --interval=all test_input
dn query --interval=all test_input
dn query --interval=all test_input -b req.method
rm -rf $tmpfile $tmpfile2
