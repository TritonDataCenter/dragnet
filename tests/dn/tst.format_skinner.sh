#!/bin/bash

#
# tst.format_skinner.sh: tests "scan", "index", and "query" operations on
# skinner-style points as input data.
#

set -o errexit
. $(dirname $0)/common.sh

function trace
{
	echo "#" "$@"
	"$@"
}

tmpfile="/var/tmp/$(basename $0).$$"
tmpfile2="$tmpfile.2"
echo "using tmpfiles \"$tmpfile\" and \"$tmpfile2\"" >&2

# Test points with no fields
dn scan --points $DN_DATADIR/2014/05-01/one.log > $tmpfile
cat $tmpfile | trace dn scan --data-format=json-skinner /dev/stdin
cat $tmpfile | trace dn scan --data-format=json-skinner /dev/stdin
cat $tmpfile $tmpfile | trace dn scan --data-format=json-skinner /dev/stdin
cat $tmpfile $tmpfile $tmpfile | \
    trace dn scan --data-format=json-skinner /dev/stdin

# Test points with a couple of fields
dn scan --points -b req.method,res.statusCode \
    $DN_DATADIR/2014/05-01/one.log > $tmpfile
dn scan -b req.method $DN_DATADIR/2014/05-01/one.log
cat $tmpfile $tmpfile $tmpfile | \
    trace dn scan --data-format=json-skinner /dev/stdin
cat $tmpfile $tmpfile $tmpfile | \
    trace dn scan -b req.method --data-format=json-skinner /dev/stdin

# Test indexes
echo "building index"
cat $tmpfile $tmpfile $tmpfile | \
    dn index -c req.method --data-format=json-skinner /dev/stdin $tmpfile2
dn query $tmpfile2
dn query -b req.method $tmpfile2

rm -f $tmpfile $tmpfile2
