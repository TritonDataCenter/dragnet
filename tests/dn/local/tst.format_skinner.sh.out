# dn scan stdin-skinner
VALUE
  250
# dn scan stdin-skinner
VALUE
  500
# dn scan stdin-skinner
VALUE
  750
REQ.METHOD VALUE
DELETE        74
GET           54
HEAD          61
PUT           61
# dn scan stdin-skinner
VALUE
  750
# dn scan stdin-skinner -b req.method
REQ.METHOD VALUE
DELETE       222
GET          162
HEAD         183
PUT          183
building index
VALUE
  750
REQ.METHOD VALUE
DELETE       222
GET          162
HEAD         183
PUT          183
